Lib/ntpath.py

```diff
@@ -179,45 +206,19 @@ def splitroot(p):
         splitroot('C:///spam///ham') == ('C:', '/', '//spam///ham')
         splitroot('Windows/notepad') == ('', '', 'Windows/notepad')
     """
-    p = os.fspath(p)
+    drive, p = splitdrive(p)
     if isinstance(p, bytes):
```

Comment:
Repeats work already done in `splitdrive()`

Reply:
@barneygale, is there a relevant discussion for this reversion to doing all the work in `splitdrive()`? I thought the plan was to implement a fast version of `splitroot()` in C, which would be used by `splitdrive()` and other functions.

Reply:
@eryksun, no. I wasn't aware of such plans. But the way it's written right now, the logic of splitting the drive and root is intertwined. Logically you need to split the drive, and the root too if the remaining path starts with a (back)slash.

Reply:
The implementations of `os.path` functions call `splitroot()`, not `splitdrive()`. This change will make all of that code slower. The extra cost to split out the root is minimal in context, but Python function calls are always expensive.

Reply:
OK, then we can also split the drive and the root in `splitroot` in 2 steps, see the updated implementation.